controllers/space/MembershipController.php

```diff
@@ -68,4 +69,51 @@ public function actionIndex($spaceId){
         }
         return $this->returnPagination($query,$pagination,$results);
     }
+
+    /**
+     * @return void
+     * Get all the spaces list of the user
+     */
+    //Get the list of all spaces of the user (issue-8 new endpoint)
+    public function actionGetSpaces(){
+       $userId = Yii::$app->user->id;
+
+       $user = User::findOne($userId);
+
+       if($user == null){
+           return $this->returnError(404, 'User not found!');
+       }
+
+       $spaceMemberships = Membership::find()->where(['user_id' => $userId]);
+
+        if($spaceMemberships->all() == null){
+            return $this->returnError(404, 'Space not found for the user : '.$userId);
+        }
+
+        $results = [];
+        $pagination = $this->handlePagination($spaceMemberships);
+
+        foreach ($spaceMemberships->all() as $spaceMembership){
+            $results[] = $this->getSpacesListData($spaceMembership);
+        }
+
+        return $this->returnPagination($spaceMemberships,$pagination,$results);
+    }
+
+    public function getSpacesListData(Membership $membership){
+        return [
+            'user' => UserDefinitions::getUserShort($membership->user),
+            'space_id' => $membership->space_id,
+            'role' => $membership->group_id,
+            'status' => $membership->status,
+            'can_cancel_membership' => $membership->can_cancel_membership,
+            'send_notifications' => $membership->send_notifications,
+            'show_at_dashboard' => $membership->show_at_dashboard,
+            'originator_user' => ($membership->originator !== null) ? UserDefinitions::getUserShort($membership->originator) : null,
+            'member_since' => $membership->created_at,
+            'request_message' => $membership->request_message,
+            'updated_at' => $membership->updated_at,
+            'last_visit' => $membership->last_visit,
```

Comment:
please add `name`,  `description`, `visibility` and `tags` of the space to be more in common with the result of `GET /space`

Reply:
a quick comparison of the responses:

![image](https://user-images.githubusercontent.com/1942953/183898669-53e07f06-d700-4e12-86d1-8bce1fc38ded.png)
(left `GET /space/memberships` right `GET /space`)

Reply:
imagine a list of all spaces that you as a user are member or admin. this list would be presented with names, descriptions, status and maybe more. to be possible to present it in this way, the information need to be returned by the endpoint. does this help?